File: app/src/main/java/com/icostel/arhitecturesample/di/modules/FragmentModule.java
package com.icostel.arhitecturesample.di.modules;

import com.icostel.arhitecturesample.ui.about.AboutFragment;
import com.icostel.arhitecturesample.ui.listusers.ListUsersFragment;
import com.icostel.arhitecturesample.ui.userdetails.UserDetailsFragment;
import com.icostel.arhitecturesample.utils.error.ErrorFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class FragmentModule {

    @ContributesAndroidInjector
    abstract ErrorFragment contributesErrorFragment();

    @ContributesAndroidInjector
    abstract ListUsersFragment contributesListUsersFragment();

    @ContributesAndroidInjector
    abstract UserDetailsFragment contributesUserDetailsFragment();

    @ContributesAndroidInjector
    abstract AboutFragment contributesAboutFragment();
}

File: app/src/main/java/com/icostel/arhitecturesample/di/modules/FragmentModule.kt
package com.icostel.arhitecturesample.di.modules

import com.icostel.arhitecturesample.ui.screens.about.AccountFragment
import com.icostel.arhitecturesample.ui.screens.listusers.ListUsersFragment
import com.icostel.arhitecturesample.ui.screens.userdetails.UserDetailsFragment
import com.icostel.arhitecturesample.utils.error.ErrorFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class FragmentModule {

    @ContributesAndroidInjector
    abstract fun contributesErrorFragment(): ErrorFragment

    @ContributesAndroidInjector
    abstract fun contributesListUsersFragment(): ListUsersFragment

    @ContributesAndroidInjector
    abstract fun contributesUserDetailsFragment(): UserDetailsFragment

    @ContributesAndroidInjector
    abstract fun contributesAccountFragment(): AccountFragment
}
